Us 51

lknaron opened the pull request:
This will perform a check of the date and time a user is trying to use to schedule an event.

If the event is scheduled as a non-repeating event on the current date AND the time trying to be scheduled is less than the user's current time, a notice will be given that the time has already occurred.

Should not affect any repeatable event or future event.

Files modified in this branch:
 EventsManager.java
 EventDialog.java
 DailyItemsPanel.java
 EventsManagerTest.java
 AllTests.java

jdhinkle commented:
| ID# | Location | Problem Description | Category | Severity |
| :-: | --- | :-: | :-: | :-: |
| A.1 | src/net/sf/memoranda/EventsManager.java: 38,412,455,515 | Improper formatting or missing banner comments according to coding standards (CG: 2) | CG | LOW |
| A.2 | src/net/sf/memoranda/EventsManager.java: 49,50,413,456,516 | Class member's scopes should be private according to coding standards (CG:5) | CG | MJ |
| A.3 | src/net/sf/memoranda/EventsManager.java: 65,84,95,104,120,142,170, 233,250,280,291,341,345,361,367 | Improper formatting or missing banner comments according to coding standards (CG: 3) | CG | LOW |
| A.4 | src/net/sf/memoranda/EventsManager.java: 372,385,392,402 | Missing underscore in front of private methods (CG:4.b) | CG | LOW |
| A.5 | src/net/sf/memoranda/EventsManager.java: 109 | Missing curly braces in loop (CG:8.c) | CG | LOW |
| A.6 | src/net/sf/memoranda/EventsManager.java: 175 | String.split throws an unchecked exception, and rethrowing the same exception seems redundant. | CS | LOW |
| A.7 | src/net/sf/memoranda/EventsManager.java: 229,231 | Unnecessary comments | MD | LOW |
| B.1 | src/net/sf/memoranda/ui/DailyItemsPanel.java: 73,75,76,78,79 | Class member's scopes should be private according to coding standards (CG:5) | CG | MJ |
| B.2 | src/net/sf/memoranda/ui/DailyItemsPanel.java: 409 | Improper formatting of banner comments according to coding standards (CG: 3) | CG | LOW |
| C.1 | src/net/sf/memoranda/ui/EventDialog.java: 83,84 | Class member's scopes should be private according to coding standards (CG:5) | CG | MJ |
| C.2 | src/net/sf/memoranda/ui/EventDialog.java: 489 | Improper formatting of banner comments according to coding standards (CG: 3) | CG | LOW |

lknaron commented:
I made some changes per your review. I didn't make any changes to original code (except for braces), only to segments I added. Below is a summary of what I did:

A.1 no changes made
A.2 no changes made
A.3 added banner to getFormattedLocalTime() method,
      fixed checkEventSchedule() banner, 
      fixed getFormattedLocalTime() banner
A.4 no changes made
A.5 added curly braces
A.6 added IllegalArgumentException
A.7 removed comments
B.1 no changes made
B.2 fixed getCurrentDate() banner
C.1 no changes made
C.2 fixed okB_actionPerformed() banner
